src/main/scala/org/apache/spark/sql/delta/DeltaOptions.scala

```diff
@@ -89,6 +90,15 @@ trait DeltaWriteOptionsImpl extends DeltaOptionParser {
     options.get(DATA_CHANGE_OPTION).map(!toBoolean(_, DATA_CHANGE_OPTION)).getOrElse(false)
   }
 
+  /** Whether to only overwrite partitions that have data written into it at runtime. */
+  def dynamicPartitionOverwriteMode: Boolean = {
+    val mode = options.get(PARTITION_OVERWRITE_MODE)
+      .map(_.toUpperCase(Locale.ROOT))
+      .getOrElse(sqlConf.getConf(SQLConf.PARTITION_OVERWRITE_MODE))
+    require(mode == "STATIC" || mode == "DYNAMIC",
```

Comment:
Why are you checking validity of the property here?

Reply:
to avoid confusion... like someone uses wrong value by accident for DYNAMIC and it silently goes back to STATIC

Reply:
Imagine calling the method only to get an exception that the value is incorrect. I think this simple method would do too much. That should be someone else's concern. Just saying as I believe it belongs somewhere else (no idea where though).

Reply:
changed it to use validOptionKeys and a separate method called validatePartitionOverwriteMode